### DISCUSSION TITLE
i want share context between layout and page but when i use sleep or async why didn't work

### DISCUSSION THREAD

**Discussion by itorz7** [remix-run/remix#10179] [Q&A]
![image](https://github.com/user-attachments/assets/407f2279-c7bd-4cc7-8aa2-a5a329d49f2d)
app/routes/app.tsx

![image](https://github.com/user-attachments/assets/8a520d8a-7b75-4eac-9530-c349d09689e1)
app/routes/admin._index.tsx

![image](https://github.com/user-attachments/assets/e11ce282-d154-4baf-9d8b-5b3833d5dd82)
log

i cannot get context but if i remove sleep i working
![image](https://github.com/user-attachments/assets/29e2346f-da9c-42d0-9fed-8077c90b2c8c)

**Comment by sergiodxa** [comment]
The context object you receive in the loaders is not intended to be changed at that point, is an object that's used by the HTTP server to inject values into the Remix app.

And since loaders run in parallel while your first loader is sleeping the second keeps working.
